tests/runtime/probe

```diff
@@ -141,6 +141,25 @@ EXPECT ERROR: Error attaching probe: kprobe:nonsense:vfs_read: specified module
 TIMEOUT 5
 WILL_FAIL
 
+NAME kprobe_wildcard_all_leading_underscore
```

Comment:
This test seems to be causing an issue in CI. I think we can probably get rid of this one as it probably just takes to long to attach. The ones below are great though - thanks for adding.

Reply:
> This test seems to be causing an issue in CI. I think we can probably get rid of this one as it probably just takes too long to attach. The ones below are great though - thanks for adding.

It looks like it's failing because it's exceeding the default maximum number of probes (512). This hasn't been an issue in testing as the kernels i work on have kprobe multi enabled and we end up reporting a small number of probes even if thousands are enabled. Victor was looking at this recently so maybe that has been fixed. I'll sort something out (either drop it or fix it).

Reply:
@jordalgo I just upped the max probe limit for that one test to 20k (totally random value really that is ~3x the current number of probes).

Reply:
I pulled down this PR locally and when I run this prog on my dev machine I get this error "ERROR: Error attaching probe: kprobe:_*". Looks like the errno is ESRCH - so not sure what's going on there.

Anyway, this looks like it passes CI just fine but I'll wait another day to see if anyone has any objections before I merge.

Reply:
Beware that attaching to too many kprobes may cause kernel hang or panic (that's why the artificial limit is there). Also that's one of the reasons why kprobe_multi was introduced, despite the name, it uses a different attachment mechanism (fprobe). I'd avoid running this test when kprobe_multi is not available on the system.

Reply:
My vote is to just remove this test but if you want to keep then you can use this to require kprobe multi `REQUIRES_FEATURE kprobe_multi`

Reply:
> Beware that attaching to too many kprobes may cause kernel hang or panic (that's why the artificial limit is there). Also that's one of the reasons why kprobe_multi was introduced, despite the name, it uses a different attachment mechanism (fprobe). I'd avoid running this test when kprobe_multi is not available on the system.

@viktormalik Good point on the kprobe_multi feature requirement - I'll add that to that test. The point of this change (and possible subsequent changes) is to make it so that a user can light up a lot of probes via globbing without having to worry about stability issues - they simply shouldn't have to especially when dealing with kernel probes entry/return probes. There may have to be a number of iterations at this but the current limitation (512 probes) is pretty ridiculous and I think fairly arbitrary from what I remember. Of course, keep a limit in for now but removing functions with known stability issues is a start in that direction.

Reply:
> I pulled down this PR locally and when I run this prog on my dev machine I get this error "ERROR: Error attaching probe: kprobe:_*". Looks like the errno is ESRCH - so not sure what's going on there.
> 
> Anyway, this looks like it passes CI just fine but I'll wait another day to see if anyone has any objections before I merge.

@jordalgo I'm seeing some seriously annoying behaviour on 6.4 based kernel that I have yet to get to the bottom of. I'll try to look at this again some time this week.

Reply:
> > Beware that attaching to too many kprobes may cause kernel hang or panic (that's why the artificial limit is there). Also that's one of the reasons why kprobe_multi was introduced, despite the name, it uses a different attachment mechanism (fprobe). I'd avoid running this test when kprobe_multi is not available on the system.
> 
> @viktormalik Good point on the kprobe_multi feature requirement - I'll add that to that test. The point of this change (and possible subsequent changes) is to make it so that a user can light up a lot of probes via globbing without having to worry about stability issues - they simply shouldn't have to especially when dealing with kernel probes entry/return probes. There may have to be a number of iterations at this but the current limitation (512 probes) is pretty ridiculous and I think fairly arbitrary from what I remember. Of course, keep a limit in for now but removing functions with known stability issues is a start in that direction.

Yes, absolutely, it's a good step towards that direction. OTOH, I believe that the limit doesn't (and shouldn't) apply to kprobe_multi so if we change bpftrace to report the actual number of attached functions (as we discussed in #3180), we'll have to make sure that the limit is not applied to kprobe_multi. AFAIK, the limit is only for kprobes (and potentially uprobes) where it still applies to this day (although it's been a while since I tried to crash my kernel with so many kprobes).